### DISCUSSION TITLE
OpenGL Shader Sample and idea

### DISCUSSION THREAD

**Discussion by BubbaRichard** [MonoGame/MonoGame#8246] [Content]
I've went through the documentation and know that sampling a texture in an OpenGL method isn't possible at present. I remembered that in HLSL you can return multiple values from the pixelshader. I tried working from this point of view in an DesktopGL project and have a second value for a heightmap be consumed by a second pass. I'm getting no where as I'm not even remember how to return multiple values or if it's possible in DesktopGL :). No code to share, anyone have an idea? Or is there another way to sample a texture in the DesktopGL vertex shader that I haven't found?

Thank you,
Bubba

**Comment by Apostolique** [comment]
I wrote this shader the other day that returns two values. <https://github.com/Apostolique/LostInTheLoo/blob/main/Content/bokeh-vertical.fx>

Used here:

https://github.com/Apostolique/LostInTheLoo/blob/f6dd5cc5db9af17c34a3f248cc679a9e6b43f769/Game/Renderer.cs#L41-L50

**Reply by BubbaRichard** [reply]
Thanks, I don't think that will work though. It's not returning a struct that I'm looking for.
I've found (in the past) some examples of using an out in the pixel shader but I don't know if it was in HLSL or not. If it was in hlsl, I believe the out could be used to get the height map (in another struct) in the first pass and passed to the vertex shader in the second pass.
